### PR TITLE
Display Temporary Security Credentials annotation immediately on cluster create

### PR DESCRIPTION
**Pull request by nathan-jenan-rancher**
This change makes is to that the temporary security credentials annotation
displays immediately on the cluster instead of only after the cluster has
finished provisioning.

Issue:
#16005

--------

Also corrects an intermittent test failure do to the condition dict having
no `message` field on it.

### CODE REVIEW

File: pkg/api/store/cluster/cluster_store.go [hunk 1]
```diff
@@ -50,6 +51,29 @@ func (r *Store) Create(apiContext *types.APIContext, schema *types.Schema, data
...
+			// This seems like it should be a map[string]string{} but supplying that causes annotations to blow up
+			// spectacularly
+			data[managementv3.ClusterFieldAnnotations] = make(map[string]interface{})
+		}
+
+		if annotations, ok := data[managementv3.ClusterFieldAnnotations].(map[string]interface{}); ok {
+			annotations[clusterstatus.TemporaryCredentialsAnnotationKey] = strconv.FormatBool(eksConfig.SessionToken != "")
+		}
+	}
+
```
Comment (StrongMonkey) [inline pkg/api/store/cluster/cluster_store.go]: There is a library in norman norman/types/values/values.go to help retrieving and setting nested map. So i think the code can be simplified as:
```
sessionToken := values.GetValue(data, managementv3.ClusterFieldAmazonElasticContainerServiceConfig, SessionTokenKey)
annotation := values.GetValue(data, managementv3.ClusterFieldAnnotations)
m := toMap(annotaton)
values.PutValue(data, managementv3.ClusterFieldAnnotations, m)
```

**Comment by StrongMonkey** [comment]
LGTM

File: pkg/api/store/cluster/cluster_store.go (hunk 1 above)
Comment (nathan-jenan-rancher) [reply pkg/api/store/cluster/cluster_store.go]: Incorporated this change into lines 53-60.

**Comment by cjellick** [comment]
LGTM

**Pull request merged**